pkg/expansion/system.go

```diff
@@ -176,10 +176,27 @@ func expandResource(obj *unstructured.Unstructured, ns *corev1.Namespace, templa
 	resource.SetUnstructuredContent(src)
 	resource.SetGroupVersionKind(resultantGVK)
 	resource.SetNamespace(ns.Name)
+	resource.SetName(mockNameForResource(obj, resultantGVK))
 
 	return resource, nil
 }
 
+// mockNameForResource returns a mock name for a resultant resource created
+// from expanding `gen`. The name will be of the form:
+// "mock-<resultant kind>-<generator name>". For example, a deployment named
```

Comment:
Do we want to prefix with the parent object's name? That would roughly approximate how deployment -> pod name generation works

Reply:
Sure. For a deployment named `nginx-deployment`, how about: `nginx-deployment-mock-pod`?